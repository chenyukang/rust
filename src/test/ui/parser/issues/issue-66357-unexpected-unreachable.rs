// The problem in #66357 was that the call trace:
//
// - parse_fn_block_decl
// - expect_or
// - unexpected
// - expect_one_of
// - expected_one_of_not_found
// - recover_closing_delimiter
//
// ended up bubbling up `Ok(true)` to `unexpected` which then used `unreachable!()`.
// error-pattern:  expected one of

fn f() { |[](* } //~ ERROR expected one of
//~^ ERROR expected one of
//~^^ ERROR mismatched closing delimiter
//~| ERROR `main` function not found
