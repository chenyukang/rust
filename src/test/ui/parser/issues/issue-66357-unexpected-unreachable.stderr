error: mismatched closing delimiter: `}`
  --> $DIR/issue-66357-unexpected-unreachable.rs:13:16
   |
LL | fn f() { |[](* }
   |             -  ^ mismatched closing delimiter
   |             |
   |             unmatched opening delimiter begins here

error: expected one of `,` or `:`, found `(`
  --> $DIR/issue-66357-unexpected-unreachable.rs:13:13
   |
LL | fn f() { |[](* }
   |             ^ expected one of `,` or `:`

error: expected one of `&`, `(`, `)`, `-`, `...`, `..=`, `..`, `[`, `_`, `box`, `mut`, `ref`, `|`, identifier, or path, found `*`
  --> $DIR/issue-66357-unexpected-unreachable.rs:13:14
   |
LL | fn f() { |[](* }
   |              ^ expected one of 15 possible tokens

error[E0601]: `main` function not found in crate `issue_66357_unexpected_unreachable`
  --> $DIR/issue-66357-unexpected-unreachable.rs:13:17
   |
LL | fn f() { |[](* }
   |                 ^ consider adding a `main` function to `$DIR/issue-66357-unexpected-unreachable.rs`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0601`.
